### PR TITLE
Remove wezterm fork from dependencie

### PR DESCRIPTION
**Pull request by dalton-oliveira**
Improves build time by removing wezterm dependency ([#8604](https://github.com/zed-industries/zed/issues/8604)).

### CODE REVIEW

**Comment by osiewicz** [comment]
I guess we don't need a release note here, as it's not an user-facing change.

File: crates/process_info/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#[derive(Debug, Clone)]
+pub struct ProcessInfo {
+    pub pid: u32,
+    pub name: String,
+    pub cwd: PathBuf,
+    pub argv: Vec<String>,
+}
+
+impl ProcessInfo {
+    // #[cfg(not(any(target_os = "macos", target_os = "linux", windows)))]
```
Comment (SomeoneToIgnore) [inline crates/process_info/src/lib.rs:15]: Need to remove that too, and maybe add some docs around, at least a module-level (`//!`) one.

File: crates/process_info/src/linux.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#![cfg(target_os = "linux")]
+use super::ProcessInfo;
+
+use std::{fs, path::PathBuf};
+
+impl ProcessInfo {
+    pub fn load(pid: u32) -> Option<Self> {
```
Comment (SomeoneToIgnore) [inline crates/process_info/src/linux.rs:7]: Honestly, somewhat confused that a linux version is relying on procfs parsing.
I understand that this is a somewhat standard in modern Linux distros, but we do have `libc` at our disposal and seem to do something very similar in `macos.rs` already, can we have something similar for Linux part too?

**Review by SomeoneToIgnore — COMMENTED** [review]
Looks nice, thank you for taking a stab on it.
Despite the amount of comments overall, those are are NITs not about the functionality per se.

The only thing that I sort of do not like is the procfs usage instead of libc for Linux?
Can we do it in a more universal way?

File: crates/process_info/src/linux.rs (hunk 2 above)
Comment (dalton-oliveira) [reply crates/process_info/src/linux.rs:7]: I'm using macos right know so I don't know how to make such a big change without being able to test properly.

Comment (SomeoneToIgnore) [reply crates/process_info/src/linux.rs:7]: There are so many ways including Docker, but sure, let's keep it going for now, and see if terminal on Linux misbehaves on merge.

**Comment by dalton-oliveira** [comment]
@SomeoneToIgnore added some humble documentation and copied tests from the fork. Please feel free to criticize any wording, missing info or irrelevant info on docs added. And thanks for reviewing

**Comment by mikayla-maki** [comment]
I seem to have trampled over this PR slightly, apologies!

**Comment by SomeoneToIgnore** [comment]
So now, we have the situation where Windows part for wezterm works and this PR would break it.

I would be fine with this for the Windows part only, but Linux part seems somewhat limited by this PR too due to procfs usage, so I'm not sure I'm convinced we should merge the PR as is.
Maybe @mikayla-maki has an opinion?

**Comment by dalton-oliveira** [comment]
@SomeoneToIgnore Hey! I missed the use of `sysinfo` for this PR. That will greatly reduce the complexity for this and we won't need the extra crate. Moving back to Draft

**Comment by dalton-oliveira** [comment]
@mikayla-maki glad that the windows fix went before this. I moved the solution with a slight change to a separate mod `pty_info.rs`. Can you and @JunkuiZhang review ?

Besides that:
- It was looking too cluttered on `terminal.rs` and thus moved the load and comparison part to a separate mod. Does the naming looks correct ?
- I has to upgrade `sysinfo` dep as the current version for some reason was not updating process name as expected. I had to do some small changes due to api change. 

@SomeoneToIgnore I force pushed because I started from scratch.

**Review by SomeoneToIgnore — APPROVED** [review]
Thank you for finding the right crate combinations, I've tested it on all platforms and it works on both Unix ones, but Windows one semi works:
telegram-cloud-photo-size-2-5442819962264737776-y

it always shows powershell.exe but never a subprocess.
I guess it's not bad already, give then Windows state, and the burden of checking out the repo is bigger so I'd like to merge it in.

**Pull request merged**